src/runtime_src/core/common/message.cpp

```diff
@@ -277,5 +279,40 @@ send(severity_level l, const char* tag, const char* msg)
     dispatcher->send(l, tag, msg);
   }
 }
+
+void
+send(severity_level l, const char* tag, const char* format, va_list args)
+{
+  static auto verbosity = xrt_core::config::get_verbosity();
+  if (l > (xrt_core::message::severity_level)verbosity) 
+    return;
+  
+  va_list args_bak;
+  // vsnprintf will mutate va_list so back it up
+  va_copy(args_bak, args);
+  int len = std::vsnprintf(nullptr, 0, format, args_bak);
+  va_end(args_bak);
+  if (len <= 0) {
+    //illegal arguments
+    std::string err_str = "ERROR: Illegal arguments in log format string. ";
+    err_str.append(std::string(format));
+    send(l, tag, err_str);
+    return;
+  }
+  ++len; //To include null terminator
+  std::vector<char> buf(len);
+  std::fill(buf.begin(), buf.end(), 0);
```

Comment:
How about `std::vector<char> buf(len, 0)`;

Reply:
this works